### PR TITLE
feat(game): add type indicators to achievements [progression]

### PR DESCRIPTION
**Pull request by wescopeland**
> ✅ Safe to merge, supports the "beat" feature flag.

This PR implements the new type indicators to the game page achievements list (https://discord.com/channels/476211979464343552/1142257722788679780/1142258578338631751).

In doing so, the achievement rows have been rewritten and migrated to Blade. Several bugs have been fixed in the process. The mobile experience has also been improved, as has accessibility.

As of right now, nothing happens when tapping on one of these type indicators. However, it would be reasonable for them to open the (unimplemented, coming in an imminent PR) beaten game credit dialog (https://discord.com/channels/476211979464343552/1142257722788679780/1142258738946916482).

## UX Walkthrough

This PR formally introduces the Progression and Win Condition iconography:
Screenshot 2023-08-25 at 9 52 13 PM

### Desktop
The layout between this PR and prod is virtually the same, with some subtle differences. I wanted to keep the layouts close to the same because I believe the current ability to quickly scroll through the table rows is very desirable and should remain unchanged.

* Points are now the text color to match the player card / new navbar, and are spaced ever so slightly away from the achievement title.
* If the achievement is unlocked, the date has a more human-friendly format.
* Points and RetroPoints will never appear on different lines.
* RetroPoints now uses the `<x-points-weighted-container />` component.
* Explanation `title` attrs have been added to the progressbar labels.
* The "[m]" tag will now never appear between points and RetroPoints.

The biggest obvious addition to the rows is the type indicator:

**Before**
Screenshot 2023-08-25 at 9 56 21 PM

**After**
Screenshot 2023-08-25 at 9 49 35 PM

On hover, these indicators expand to reveal their labels:
Screenshot 2023-08-25 at 9 51 09 PM

### Mobile
Mobile has been redesigned.

**Before**
Screenshot 2023-08-25 at 9 55 05 PM

**After**
Screenshot 2023-08-25 at 9 54 30 PM

This was the original design that was planned for desktop, however after trying it in real life beyond just the wireframes, I wasn't happy with how it turned out. On mobile devices, it works quite nicely. The text being above the progress bar makes it easier to connect what label is associated with the bar while quickly skimming through the list.

Tapping on the type indicator does nothing, however this will be addressed with the upcoming beaten game credit dialog PR.

### Other nuance
Light mode has special styles to ensure it's supported. This PR cherry-picks the custom light mode Tailwind plugin from #1785.

Screenshot 2023-08-25 at 9 59 25 PM

---

https://github.com/RetroAchievements/RAWeb/assets/3984985/2a0f28b7-6a25-4fe6-8723-b6f66921b357

https://github.com/RetroAchievements/RAWeb/assets/3984985/69804d52-fe5c-4841-ad2c-1b512e0927ca

### CODE REVIEW

**Comment by Jamiras** [comment]
Most win conditions don't seem to be working. Game 697:
image

image

Game 14518:
image

image

Game 6346:
image

image

**Comment by Jamiras** [comment]
Also (unrelated to the functionality), I'm not sure how progression achievements are being determined currently, but you can't use something unless you obtain it first:
image

**Comment by wescopeland** [comment]
>Most win conditions don't seem to be working.

If a game only has one Win Condition, to confuse players less, it's rendered as Progression. Is that currently what you're seeing?

**Comment by Jamiras** [comment]
> If a game only has one Win Condition, to confuse players less, it's rendered as Progression. Is that currently what you're seeing?

Yep, I made that realization and updated the original comment.

File: resources/views/platform/components/game/achievements-list/list-item-global-progress.blade.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+?>
+
+
+<p class="text-2xs text-center hidden md:block -mt-1.5">
+    {{ number_format($unlockRate, 2) }}% unlock rate
+</p>
+
+<p id="progress-label-{{ $achievement['ID'] }}" class="mb-0.5 text-2xs md:text-center md:mb-0">
+    <span title="Total unlocks" class="cursor-help">{{ localized_number($wonBy) }}</span>
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply resources/views/platform/components/game/achievements-list/list-item-global-progress.blade.php:39]: Resolved in https://github.com/RetroAchievements/RAWeb/pull/1802/commits/3a95d457d4108efcb0e65c4a50662b4b001add89.

**Review by Jamiras — APPROVED** [review]

**Comment by wescopeland** [comment]
This PR now has added support for the new "beat" feature flag and can be merged at will.

When the flag is not enabled, the type indicators are not rendered in the UI.

**Comment by wescopeland** [comment]
Per community feedback, I've ripped out the code that would mask a single Win Condition achievement as being Progression.

**Pull request merged**